packages/mui-material/src/Autocomplete/Autocomplete.spec.tsx

```diff
@@ -171,3 +172,22 @@ function CustomListboxRef() {
     >(e);
   }}
 />;
+
+function AutocompleteCustomPaper() {
+  return (
+    <Autocomplete
+      options={['one', 'two', 'three']}
+      renderInput={(params) => <TextField {...params} />}
+      // eslint-disable-next-line react/no-unstable-nested-components
+      PaperComponent={({ elevation, variant, ...props }) => (
+        <Paper {...{ elevation, variant }} {...props} />
+      )}
```

Comment:
Rendering the component inline isn't recommended due to the eslint rule `react/no-unstable-nested-components`. However, it seems we can't test it any other way. Let's keep the `PaperComponent` type modification but remove this test. Developers can render a custom Paper component like this:
```tsx
function CustomPaper(props: PaperProps) {
  return <Paper square {...props} />;
}

function AutocompleteCustomPaper() {
  return (
    <Autocomplete
      options={['one', 'two', 'three']}
      renderInput={(params) => <TextField {...params} />}
      PaperComponent={CustomPaper}
      componentsProps={{
        paper: {
          elevation: 2,
          variant: 'outlined',
        },
      }}
    />
  );
}
```
It seems okay not to have a test for the modification of the `PaperComponent` prop type because, apart from inlining a component (not recommended), we can't test it. In any case, we plan to deprecate this approach and use the `slots` prop instead (`slots.paper`).